pandas/core/arrays/base.py

```diff
@@ -653,15 +660,60 @@ def copy(self, deep=False):
         raise AbstractMethodError(self)
 
     # ------------------------------------------------------------------------
-    # Block-related methods
+    # Printing
     # ------------------------------------------------------------------------
 
+    def __repr__(self):
+        from pandas.io.formats.printing import format_object_summary
+
+        template = (
+            '<{class_name}>\n'
+            '{data}\n'
+            'Length: {length}, dtype: {dtype}'
```

Comment:
do we need to define the “__unicode__”
we do this in Base for all pandas objects

Reply:
you are writing new code here but this should be consistent as well (it’s ok to change that too)
but to have a completely different impl is odd

Reply:
Fixed... I left the implementation in repr, and then encoded / decoded as needed in `__unicdoe__` and `__bytes__` if that's OK.